### ISSUE TITLE
improving control flow for lb_v2 resources

### ISSUE THREAD

**Issue by jrperritt** [terraform-providers/terraform-provider-openstack#4]
The control flow of waiting for LB v2 resources to become ready to create/update could be improved. Currently, for example, it just immediately retries if it gets a `409`, which can lead to what appear to be race conditions in OpenStack. It seems like the problem should be fixed in OpenStack, but Terraform should handle this better by, for example, waiting until the LB is `ACTIVE` before continuing to try to create the LB pool after receiving a `409`.

### Terraform Version
0.9.8.

### Affected Resource(s)
- openstack_lb_*_v2

### Terraform Configuration Files

You'll need to provide `image_id` and `flavor_id`:

```hcl
variable "subnet_id" {
  type = "string"
  default = "4409be68-f1d6-44a5-a54a-bc22e790b237"
}

variable "image_id" {
  type = "string"
}

variable "flavor_id" {
  type = "string"
}

resource "openstack_lb_loadbalancer_v2" "lb" {
  vip_subnet_id = "${var.subnet_id}"
  name = "lb"
}

resource "openstack_lb_listener_v2" "lb_listener" {
  loadbalancer_id = "${openstack_lb_loadbalancer_v2.lb.id}"
  protocol = "HTTPS"
  protocol_port = 443
}

resource "openstack_lb_pool_v2" "lb_pool" {
  lb_method = "ROUND_ROBIN"
  protocol = "HTTPS"
  listener_id = "${openstack_lb_listener_v2.lb_listener.id}"
}

resource "openstack_lb_member_v2" "lb_member" {
  address = "${openstack_compute_instance_v2.server.access_ip_v4}"
  pool_id = "${openstack_lb_pool_v2.lb_pool.id}"
  protocol_port = 443
  subnet_id = "${var.subnet_id}"
}

resource "openstack_lb_monitor_v2" "lb_monitor" {
  delay = 30
  max_retries = 3
  pool_id = "${openstack_lb_pool_v2.lb_pool.id}"
  timeout = 5
  type = "PING"
}

resource "openstack_compute_instance_v2" "server" {
  name      = "server"
  image_id  = "${var.image_id}"
  flavor_id = "${var.flavor_id}"
  config_drive = false
}
```

### Expected Behavior
1 LB pool gets created

### Actual Behavior
4 LB pools get created

### Steps to Reproduce
1. `terraform apply`

**Comment by jtopjian** [comment]
@jrperritt Is this similar to https://github.com/hashicorp/terraform/pull/15225 or is a 409 being hit elsewhere?

I should probably just blanket all LB resources with 409 checks.

**Comment by jrperritt** [comment]
Yes, similar to that, but the fix that was implemented is not sufficient. Instead of immediately retrying the request, `waitForLoadBalancerActive` (or similar) should be called to wait for an `ACTIVE` state.

**Comment by jrperritt** [comment]
I'll submit a fix for this.

**Comment by jtopjian** [comment]
#25 has been merged.

**Comment by jtopjian** [comment]
Fixed by #29
